pkg/asset/manifests/registry.go

```diff
@@ -0,0 +1,104 @@
+package manifests
+
+import (
+	"fmt"
+	"path/filepath"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+
+	imageregistryv1 "github.com/openshift/api/imageregistry/v1"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/azure"
+)
+
+var clusterImageRegistryConfigFile = filepath.Join(manifestDir, "cluster-image-registry-02-config.yml")
+
+// ImageRegistry generates the cluster-ingress-*.yml files.
+type ImageRegistry struct {
+	FileList []*asset.File
+}
+
+var _ asset.WritableAsset = (*ImageRegistry)(nil)
+
+// Name returns a human friendly name for the asset.
+func (*ImageRegistry) Name() string {
+	return "Image Registry Config"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*ImageRegistry) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+	}
+}
+
+// Generate generates the internal image registry cluster config.
+//
+// A cluster image registry config is only created for Azure when the cluster
+// is using an internal publishing strategy.
+func (ing *ImageRegistry) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	dependencies.Get(installConfig)
+
+	config := installConfig.Config
+	ing.FileList = []*asset.File{}
+
+	clusterConfig, err := ing.generateClusterConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create cluster config for internal image registry: %w", err)
+	}
+	ing.FileList = append(ing.FileList, &asset.File{
+		Filename: clusterImageRegistryConfigFile,
+		Data:     clusterConfig,
+	})
+
+	return nil
+}
+
+func (ing *ImageRegistry) generateClusterConfig(config *types.InstallConfig) ([]byte, error) {
+	if config.Platform.Name() != azure.Name || config.Publish != types.InternalPublishingStrategy {
+		return []byte{}, nil
+	}
+
+	obj := &imageregistryv1.Config{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: imageregistryv1.GroupVersion.String(),
+			Kind:       "Config",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+			// not namespaced
+
+			// the registry operator will not generate its own default config
+			// unless we explicitly tell it that this config was generated by us.
+			// note that only the storage part of this is used by the registry
+			// operator, and only for Azure.
+			Annotations: map[string]string{"installer-bootstrapped": ""},
```

Comment:
The registry operator bootstrap code takes an all or nothing approach to configuring the CR: https://github.com/openshift/cluster-image-registry-operator/blob/master/pkg/operator/bootstrap.go#L33-L36.
I added this annotation as a way to tell it that the installer is providing this config for storage only, so that the operator can safely use that when bootstrapping itself.
I'm open to suggestions on both naming and different approaches to achieve this.
@dmage you may have an opinion here.

Reply:
Link to registry operator PR: https://github.com/openshift/cluster-image-registry-operator/pull/962